### DISCUSSION TITLE
Local development with https

### DISCUSSION THREAD

**Discussion by charbs-io** [pocketbase/pocketbase#3161] [Q&A] [answered]
I cannot figure out how i can start local development with the --https flag.

I'm trying something like `./pocketbase serve --https 127.0.0.1:8090` but get `http: TLS handshake error from 127.0.0.1:56294: acme/autocert: server name component count invalid` when i try accessing the api or admin locally.

Can anyone help? I'm trying to test a web app on mobile on the same network but i'm denied from making calls to an http server.

**Comment by ganigeorgiev** [comment, marked as answer]
The `--https` flag issue a Let's Encrypt certificate and it requires a valid public domain name in order to work.

We don't support issuing or loading self-signed certificates out of the box (~and even if there was such option~_edit: with the latest releases you have access to the `e.Server` and `e.TLSManager` fields as part of the `OnServe` hook_, you still will have to import it manually on the device where you are testing since it won't be signed by a common trusted CA).

If you have a requirement to use TLS (_eg. required by some OAuth2 vendor_), my recommendation would be to use something like [ngrok](https://ngrok.com/) (eg. `ngrok http 8090`) that will create a public TLS address for your localhost and which you can access on your mobile app even if the device is not in the same network.

**Reply by charbs-io** [reply]
Thanks @ganigeorgiev - thought ngrok was an option, but good to confirm there's no support out of the box.

**Reply by ganigeorgiev** [reply]
Tunneling services like ngrok are the easiest option in my opinion.
TLS on localhost across devices is usually tricky and I'm not sure if there is an easy and automated way to do it.
You can also check https://letsencrypt.org/docs/certificates-for-localhost/.

**Reply by florianrusch** [reply]
An alternative to manually creating a self-signed certificate would be to use https://github.com/FiloSottile/mkcert. But even this would require you to somehow load the certificate into pocketbase.

**Reply by bachgg** [reply]
@charbs-io hey, if you still need this, I wrote an instruction [here](https://github.com/moppediert/localssl). The gist of it is to point a domain name that you own to the local IP address of your development machine and use traefik for automating certificate request using the DNS challenge. Then you'll get a valid cert from letsencrypt and other devices in the same network can access your server over https. I needed this for a websocket server on my computer that a phone needs to connect to over https (or rather wss). Hope this helps.
